src/pymor/models/iosys.py

```diff
@@ -928,6 +929,45 @@ def get_ast_spectrum(self, ast_pole_data=None, mu=None):
 
             return ast_lev, ast_ews[idx], ast_rev
 
+    def moebius_substitution(self, M, sampling_time=0):
+        assert isinstance(M, MoebiusTransform)
+        if not isinstance(self.E, IdentityOperator):
+            return NotImplemented
+
+        a, b, c, d = M.coefficients
+        s = a * d - b * c
+        v = np.sqrt(np.abs(s))
+        Gamma = NumpyMatrixOperator(
+            spla.inv(to_matrix(a * self.E - c * self.A)),
+            source_id=self.A.source_id,
+            range_id=self.A.range_id)
+
+        # At = to_matrix(Gamma @ (d * self.A - b * self.E))
+        At = M.apply_inverse(M.source.from_numpy(self.A.matrix)).to_numpy().T
+        Bt = to_matrix(np.sign(s) * v * Gamma @ self.B)
+        Ct = to_matrix(v * self.C @ Gamma)
+        Dt = to_matrix(self.D + c * self.C @ Gamma @ self.B)
```

Comment:
`to_matrix` should be avoided, unless this is expected to be used only for small models.